eth-node/types/rpc.go

```diff
@@ -6,19 +6,20 @@ import (
 
 // NewMessage represents a new whisper message that is posted through the RPC.
 type NewMessage struct {
-	SymKeyID    string    `json:"symKeyID"`
-	PublicKey   []byte    `json:"pubKey"`
-	SigID       string    `json:"sig"`
-	TTL         uint32    `json:"ttl"`
-	PubsubTopic string    `json:"pubsubTopic"`
-	Topic       TopicType `json:"topic"`
-	Payload     []byte    `json:"payload"`
-	Padding     []byte    `json:"padding"`
-	PowTime     uint32    `json:"powTime"`
-	PowTarget   float64   `json:"powTarget"`
-	TargetPeer  string    `json:"targetPeer"`
-	Ephemeral   bool      `json:"ephemeral"`
-	Priority    *int      `json:"priority"`
+	SymKeyID             string    `json:"symKeyID"`
+	PublicKey            []byte    `json:"pubKey"`
+	SigID                string    `json:"sig"`
+	TTL                  uint32    `json:"ttl"`
+	PubsubTopic          string    `json:"pubsubTopic"`
+	Topic                TopicType `json:"topic"`
+	Payload              []byte    `json:"payload"`
+	Padding              []byte    `json:"padding"`
+	PowTime              uint32    `json:"powTime"`
+	PowTarget            float64   `json:"powTarget"`
+	TargetPeer           string    `json:"targetPeer"`
+	Ephemeral            bool      `json:"ephemeral"`
+	Priority             *int      `json:"priority"`
+	ContentTopicOverride string    `json:"contentTopicOverride"`
```

Comment:
This looks quite counter-intuitive. Why do we want some special `ContentTopicOverride` field, and not just use `Topic`?

Reply:
good point, let me think of why i did not do that. maybe it is possible.

but for some reference as to why i had to use a separate override field
-  https://github.com/status-im/status-go/pull/5864/files#r1825630510

Reply:
I had to do this as https://github.com/status-im/status-go/blob/d61d20042aa8956ce8634da0ee5f603df379d6dc/protocol/messenger.go#L2272 doesn't have access for logic to derive content-topic from chatID and similarly https://github.com/status-im/status-go/blob/d61d20042aa8956ce8634da0ee5f603df379d6dc/protocol/common/message_sender.go#L668.

Happy to follow any suggestion you have.

Reply:
Thanks to @osmaczko changes in https://github.com/status-im/status-go/pull/6179, now this is lot cleaner.